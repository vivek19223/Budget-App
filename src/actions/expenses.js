import {v4 as uuid} from 'uuid';

//Create addExpense
export const addExpense = (
  {description = '', note = '', amount = 0, createdAt = 0} = {}
) => ({
  type: 'ADD_EXPENSE',
  expense: {
    id: uuid (),
    description,
    note,
    amount,
    createdAt,
  },
});

//Create removeExpense
export const removeExpense = ({id}) => ({
  type: 'REMOVE_EXPENSE',
  id,
});

//Create editExpense
export const editExpense = (id, updates) => ({
  type: 'EDIT_EXPENSE',
  id,
  updates,
});
