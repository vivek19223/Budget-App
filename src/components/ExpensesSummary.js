import React from 'react';
import ExpenseTotal from '../selectors/expense-total';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral'

export const ExpensesSummary = props => {
    const total = numeral(ExpenseTotal (props.expenses)).format('0,0')
    return (
        <div>
            {props.expenses.length > 1
            ? <h1>Viewing {props.expenses.length} expenses totalling {total}</h1>
            : props.expenses.length === 1 &&
             <h1>Viewing {props.expenses.length} expense totalling {total} </h1>}
        </div>
    )
};

const mapStateToProps = state => {
  return {
    expenses: selectExpenses (state.expenses, state.filters),
  };
};
export default connect (mapStateToProps) (ExpensesSummary);
