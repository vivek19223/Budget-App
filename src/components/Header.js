import React from 'react'
import {   NavLink } from 'react-router-dom';

const Header = ()=>(
    <header>
        <h1>Expensify</h1>
        <div>
        <NavLink to="/" activeClassName="is-active" exact={ true }>Dashboard</NavLink><br />
        <NavLink to = "/create" activeClassName="is-active">Create Expense</NavLink><br />
        </div>
    </header>
    );

export default Header;