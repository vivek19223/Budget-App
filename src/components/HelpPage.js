import React from 'react';

const HelpPage = () => (
    <div>
      This is a help component.
    </div>
  );

export default HelpPage;