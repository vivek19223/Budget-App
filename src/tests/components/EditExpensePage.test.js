import React from 'react';
import {shallow} from 'enzyme';
import {EditExpensePage} from '../../components/EditExpensePage';
import expenses from '../fixtures/expenses';

let wrapper, history, editExpense, removeExpanse;

beforeEach (() => {
  history = {push: jest.fn ()};
  editExpense = jest.fn ();
  removeExpanse = jest.fn ();
  wrapper = shallow (
    <EditExpensePage
      editExpense={editExpense}
      removeExpanse={removeExpanse}
      history={history}
      expense = {expenses[0]}
    />
  );
});

test ('test EditExpensePage correctly', () => {
  expect (wrapper).toMatchSnapshot ();
});

test ('Should Test editExpense correctly', () => {
  wrapper.find ('ExpenseForm').prop ('onSubmit') (expenses[0]);
  expect (history.push).toHaveBeenLastCalledWith ('/');
  expect(editExpense).toHaveBeenCalledWith(expenses[0].id,expenses[0])
});

// test ('Should Test removeExpense correctly', () => {
//     wrapper.find ('button').simulate ('click');
//     expect (history.push).toHaveBeenLastCalledWith ('/');
//     // expect(removeExpense).toHaveBeenLastCalledWith({
//     //     id : expenses[0].id
//     // })
//   });
  