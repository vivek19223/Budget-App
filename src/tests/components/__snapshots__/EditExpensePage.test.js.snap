// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test EditExpensePage correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 11195,
        "createdAt": 0,
        "description": "Gum",
        "id": 0,
        "note": "",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
